src/awkward/contents/listoffsetarray.py

```diff
@@ -1975,6 +1975,35 @@ def _completely_flatten(self, nplike, options):
             flat = self._content[self._offsets[0] : self._offsets[-1]]
             return flat._completely_flatten(nplike, options)
 
+    def _drop_none(self):
+        if self._content.is_option:
```

Comment:
An `IndexedOptionArray` inside of an `IndexedArray` should not appear in user arrays; these things can exist temporarily in our internal code, but before returning it to the user, `simplify_optiontype` should have turned this into a single `IndexedOptionArray`.

```
>>> layout.content
<IndexedArray len='3'>
    <index><Index dtype='int64' len='3'>[0 1 2]</Index></index>
    <content><IndexedOptionArray len='3'>
        <index><Index dtype='int64' len='3'>[-1 -1  0]</Index></index>
        <content><NumpyArray dtype='int64' len='3'>[0 1 2]</NumpyArray></content>
    </IndexedOptionArray></content>
</IndexedArray>
>>> layout.content.simplify_optiontype()
<IndexedOptionArray len='3'>
    <index><Index dtype='int64' len='3'>[-1 -1  0]</Index></index>
    <content><NumpyArray dtype='int64' len='3'>[0 1 2]</NumpyArray></content>
</IndexedOptionArray>
```

@ioanaif and I talked about this, about the implementation only needing to work for these "well formed" arrays. (We need a good name for these different levels of validity.) If we ever encounter a situation in which a user finds that `drop_none` isn't working because they have an `IndexedArray` of `IndexedOptionArray`, we'd investigate how they ended up with the `IndexedArray` of `IndexedOptionArray`, because that's already wrong.